src/coreComponents/constitutive/solid/SolidBase.hpp

```diff
@@ -558,13 +590,17 @@ class SolidBase : public constitutive::ConstitutiveBase
     static constexpr char const * defaultDensityString() { return "defaultDensity"; }  ///< Default density key
     static constexpr char const * thermalExpansionCoefficientString() { return "thermalExpansionCoefficient"; } // Thermal expansion
                                                                                                                 // coefficient key
-    static constexpr char const * defaultThermalExpansionCoefficientString() { return "defaultDrainedLinearTEC"; } // Default
-                                                                                                                   // drained
-                                                                                                                   // linear
-                                                                                                                   // thermal
-                                                                                                                   // expansion
-                                                                                                                   // coefficient
-                                                                                                                   // key
+    static constexpr char const * defaultThermalExpansionCoefficientString() { return "defaultDrainedTEC"; } // Default
+                                                                                                             // drained
+                                                                                                             // thermal
+                                                                                                             // expansion
+                                                                                                             // coefficient
+                                                                                                             // key
+
+    static constexpr char const * dThermalExpansionCoefficient_dTemperatureString() { return "dDrainedTEC_dT"; }
+    static constexpr char const * referenceTemperatureString() { return "referenceTemperature"; }
+
+    static constexpr char const * drainedTECTableNameString() { return "drainedTECTableName"; }
```

Comment:
I would add a validator here. To ensure that if the user provides a table name, then the relevant `TableFunction` exists.